File: Top 100 Codes/06-Advanced Codes related to Arrays/09-Merge 2 sorted arrays without using extra space/Merge2sortedarrayswithoutusingextraspace.py
def merge(arr1, arr2):
    n, m = len(arr1), len(arr2)
    for i in range(n):
        if arr1[i] > arr2[0]:
            # Swap if arr1[i] > arr2[0]
            arr1[i], arr2[0] = arr2[0], arr1[i]
            # Re-sort arr2 to maintain its sorted order
            arr2.sort()

# Example usage
arr1 = [1, 4, 7, 8, 10]
arr2 = [2, 3, 9]

merge(arr1, arr2)

print("Merged arrays:")
print("arr1:", arr1)
print("arr2:", arr2)

File: Top 100 Codes/06-Advanced Codes related to Arrays/09-Merge 2 sorted arrays without using extra space/Merge2sortedarrayswithoutusingextraspace.cpp
#include <iostream>
#include <algorithm> // For sort() and swap()
using namespace std;

void merge(int arr1[], int arr2[], int n, int m) {
    for (int i = 0; i < n; i++) 
    {
        if (arr1[i] > arr2[0]) 
        {
            swap(arr1[i], arr2[0]);
            sort(arr2, arr2 + m);
        }
    }
}

int main() {
    int arr1[] = {1, 4, 7, 8, 10};
    int arr2[] = {2, 3, 9};
    int n = sizeof(arr1) / sizeof(arr1[0]);
    int m = sizeof(arr2) / sizeof(arr2[0]);

    merge(arr1, arr2, n, m);

    cout << "Merged arrays:\n";
    for (int i = 0; i < n; i++){
        cout << arr1[i] << " ";
    } 
    for (int i = 0; i < m; i++){
        cout << arr2[i] << " ";
    } 
    cout << endl;

    return 0;
}
